#!/usr/bin/env python
PACKAGE = "navigation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("kp_x",     double_t,     0, "kp_x parameter", 0.0, -10.0, 10.0)
gen.add("kp_y",     double_t,     0, "kp_y parameter", 0.0, -10.0, 10.0)
gen.add("kp_z",     double_t,     0, "kp_z parameter", 0.0, -10.0, 10.0)
gen.add("kp_roll",  double_t,     0, "kp_x parameter", 0.0, -10.0, 10.0)
gen.add("kp_pitch", double_t,     0, "kp_y parameter", 0.0, -10.0, 10.0)
gen.add("kp_yaw",   double_t,     0, "kp_z parameter", 0.0, -10.0, 10.0)

gen.add("kd_x",     double_t,     0, "kd_x parameter", 1.0, -10.0, 10.0)
gen.add("kd_y",     double_t,     0, "kd_y parameter", 0.0, -10.0, 10.0)
gen.add("kd_z",     double_t,     0, "kd_z parameter", 0.0, -10.0, 10.0)
gen.add("kd_roll",  double_t,     0, "kd_x parameter", 0.0, -10.0, 10.0)
gen.add("kd_pitch", double_t,     0, "kd_y parameter", 0.0, -10.0, 10.0)
gen.add("kd_yaw",   double_t,     0, "kd_z parameter", 0.5, -10.0, 10.0)

gen.add("lambda_x",     double_t,     0, "lambda_x parameter", 0.5, -10.0, 10.0)
gen.add("lambda_y",     double_t,     0, "lambda_y parameter", 0.1, -10.0, 10.0)
gen.add("lambda_z",     double_t,     0, "lambda_z parameter", 0.0, -10.0, 10.0)
gen.add("lambda_roll",  double_t,     0, "lambda_x parameter", 0.5, -10.0, 10.0)
gen.add("lambda_pitch", double_t,     0, "lambda_y parameter", 0.1, -10.0, 10.0)
gen.add("lambda_yaw",   double_t,     0, "lambda_z parameter", 0.0, -10.0, 10.0)

gen.add("bd_x",     double_t,     0, "bd_x parameter", 0.0, -10.0, 10.0)
gen.add("bd_y",     double_t,     0, "bd_y parameter", 0.0, -10.0, 10.0)
gen.add("bd_z",     double_t,     0, "bd_z parameter", 0.0, -10.0, 10.0)
gen.add("bd_roll",  double_t,     0, "bd_x parameter", 0.0, -10.0, 10.0)
gen.add("bd_pitch", double_t,     0, "bd_y parameter", 0.0, -10.0, 10.0)
gen.add("bd_yaw",   double_t,     0, "bd_z parameter", 0.0, -10.0, 10.0)

gen.add("slave_min_x", double_t, 0, "slave_min_x parameter", -5.0, -10.0, 10.0)
gen.add("slave_max_x", double_t, 0, "slave_max_x parameter",  5.0, -10.0, 10.0)
gen.add("slave_min_y", double_t, 0, "slave_min_y parameter", -5.0, -10.0, 10.0)
gen.add("slave_max_y", double_t, 0, "slave_max_y parameter",  5.0, -10.0, 10.0)
gen.add("slave_min_z", double_t, 0, "slave_min_z parameter", -5.0, -10.0, 10.0)
gen.add("slave_max_z", double_t, 0, "slave_max_z parameter",  5.0, -10.0, 10.0)

exit(gen.generate(PACKAGE, "navigation", "SlaveController"))
